consensus/src/consensus_observer/observer.rs

```diff
@@ -881,6 +908,66 @@ impl ConsensusObserver {
             })
     }
 
+    /// Verifies the ordered blocks and returns an error if the data is invalid.
+    /// Note: this does not check the ordered proof.
+    fn verify_ordered_blocks(
```

Comment:
I don't see we use the `self` in the function, generally I think it's better to implement a `verify` function on the message instead of doing it inside the processor (for self-contained checks)

Reply:
SGTM. Will update 😄